src/operation-node/add-index-node.ts

```diff
@@ -0,0 +1,45 @@
+import { freeze } from '../util/object-utils.js'
+import { IdentifierNode } from './identifier-node.js'
+import { OperationNode } from './operation-node.js'
+import { RawNode } from './raw-node.js'
+
+export type AddIndexNodeProps = Omit<AddIndexNode, 'kind' | 'name'>
+
+export interface AddIndexNode extends OperationNode {
+  readonly kind: 'AddIndexNode'
+  readonly name: IdentifierNode
+  readonly columns?: OperationNode[]
+  readonly unique?: boolean
+  readonly using?: RawNode
+  readonly ifNotExists?: boolean
+}
+
+/**
+ * @internal
+ */
+export const AddIndexNode = freeze({
+  is(node: OperationNode): node is AddIndexNode {
+    return node.kind === 'AddIndexNode'
+  },
+
+  create(index: Omit<AddIndexNode, 'kind'>): AddIndexNode {
```

Comment:
Is there ever a situation when it's created with anything but a name at first?
If not, accept a single argument `name` here.